src/frontend/src/helpers/crypto.ts

```diff
@@ -1,3 +1,58 @@
+import * as sjcl from "sjcl";
+export {Coder}
+class Coder {
+  private cipher;
+
+  constructor(key) {
+    this.cipher = new sjcl.cipher.aes(key);
+  }
+
+  public encrypt(frame, plain) {
+    var iv = sjcl.random.randomWords(12); // We'll generate 48 bytes of entropy and use 12.
+    var ivArr = this.toBits(iv);
+    var frameArr = this.toBits(frame);
+    var plainArr = this.toBits(plain);
+    var authArr = sjcl.bitArray.concat(ivArr, plainArr);
+    var out = sjcl.mode.gcm.encrypt(this.cipher,
+        frameArr,
+        ivArr,
+        authArr,
+        128);
+    var tagArr = sjcl.bitArray.bitSlice(out, sjcl.bitArray.bitLength(out) - 128);
+    var codedArr = sjcl.bitArray.bitSlice(out, 0, sjcl.bitArray.bitLength(out) - 128);
+
+    return [this.fromBits(ivArr), this.fromBits(codedArr), this.fromBits(tagArr)];
+  }
+
+  private toBits(bytes) {
+    var out = [], i, tmp=0;
+    for (i=0; i<bytes.length; i++) {
+      tmp = tmp << 8 | bytes[i];
+      if ((i&3) === 3) {
+      out.push(tmp);
+      tmp = 0;
+      }
+    }
+    if (i&3) {
+      out.push(sjcl.bitArray.partial(8*(i&3), tmp));
+    }
+    return out;
+  }
+
+  private fromBits(arr) {
+    var out = [], bl = sjcl.bitArray.bitLength(arr), i, tmp;
+    for (i=0; i<bl/8; i++) {
+      if ((i&3) === 0) {
+        tmp = arr[i/4];
+      }
+      out.push(tmp >>> 24);
+      tmp <<= 8;
+    }
+    return out;
+  }
+}
+
+/*
```

Comment:
As far as I can tell, there's something messed up with SJCL and how it interacts with our test environment here, as it's producing incorrect output.  You should dump SJCL anyways to use the browser's native crypto functions.